### PR TITLE
Added Android dependencies and building to ci script

### PR DESCRIPTION
**Pull request by DanailPenev**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Adding build steps for `AndroidClient`, as well as a reference to the preparation script from `shared-ci`.

#### Tests
How to test the preparation script from `shared-ci` repo:
- Checkout this branch
- Run `ci/bootstrap.sh`
- Go in `.shared-ci` and checkout the `feature/add-android-dependencies-unity` branch
- Comment out the call to bootstrap in `<ROOT_DIR>/ci/build-test.sh` (this would reset back to master)
- Run `ci/build-test.sh`
- ???
- Profit

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: ci/build-test.sh [hunk 1]
```diff
@@ -7,7 +7,9 @@ cd "$(dirname "$0")/../"
 # Get shared CI and prepare unity
 ci/bootstrap.sh
 .shared-ci/scripts/prepare-unity.sh
+.shared-ci/scripts/prepare-unity-mobile.sh "$(pwd)/logs/PrepareUnityMobile.log"
 
 ci/test.sh
 .shared-ci/scripts/build.sh "workers/unity" UnityClient local "$(pwd)/logs/UnityClientBuild.log"
 .shared-ci/scripts/build.sh "workers/unity" UnityGameLogic cloud "$(pwd)/logs/UnityGameLogicBuild.log"
+.shared-ci/scripts/build.sh "workers/unity" AndroidClient local "$(pwd)/logs/AndroidClientBuild.log"
```
Comment (jessicafalk) [inline ci/build-test.sh]: need to either
1. add a check if thsi is running on a windows agent
2. make sure that we can also run it in ci for mac.

would prefer 2, but that could be enabled later

File: ci/build-test.sh [hunk 2]
```diff
@@ -7,13 +7,18 @@ cd "$(dirname "$0")/../"
 # Get shared CI and prepare unity
 ci/bootstrap.sh
 .shared-ci/scripts/prepare-unity.sh
+.shared-ci/scripts/prepare-unity-mobile.sh "$(pwd)/logs/PrepareUnityMobile.log"
```
Comment (gkassabli) [inline ci/build-test.sh]: Should we execute this only for Windows?

Comment (DanailPenev) [reply ci/build-test.sh]: Yup, good point.

Comment (jessicafalk) [reply ci/build-test.sh]: we probably do want to run it for both mac and windows at some point. but seeing as we are very limited with the number of mac agents we have and mac is kind of rekt anyway. we can isolate it to windows for now

Comment (jessicafalk) [reply ci/build-test.sh]: decided to execute it for both as it is possible for only one platform to fail

**Pull request merged**